src/Common/src/Interop/Windows/sspicli/NegotiationInfoClass.cs

```diff
@@ -17,14 +17,21 @@ internal class NegotiationInfoClass
 
         internal NegotiationInfoClass(SafeHandle safeHandle, int negotiationState)
         {
+            bool globalLogEnabled = GlobalLog.IsEnabled;
```

Comment:
I personally don't like this pattern as it adds unnecessary code complexity. 
Is there a way to redesign IsEnabled so that the compiler can make this optimization instead?